src/main/java/group1/mips_simulator/components/cpuParts/RegisterFile.java

```diff
@@ -12,7 +12,7 @@ public class RegisterFile {
     Register[] IXR = new Register[3];   //Index Reg: X1-X3.
     Register PC = new Register();       //Prog Counter Reg : address of next instruction to be executed
     Register CC = new Register();       //Condition Code Reg : set when arith/logical operations executed
```

Comment:
Typically, variable names are lowercase. So it would be `pc` instead of `cc`. 

Also, for java particularly, the language style usually prefers to be more verbose with variable names. So a typical variable here (for the java language) would be `conditionCode`, or `porgramCounter`. 

Different languages have different styles, like C is very big on acronyms and dropping vowels out of variable names, `pc` and `cc` would be more standard variable names for C.

Reply:
i totally getchu, 
1. for the naming thing, ill swap them all down to lower case,
2. and agreed this would not fly in industry, since java is more verbose but *see reasoning below*

I mentioned over text the first thing informing my style decisions is implementing hardware as close to the hardware as possible;

but the second, and more important thing for this project, is implementing as closely to the spec as possible; including naming things what the spec names them, operating on them as close to how the spec tells us to operate on them; if it mentions using a register by its acronym, we use that register by its acronym

This way, any of us can pick up and part of what the others worked on, and if they run into confusion, we can first check the spec and see how close an implementation is fitting what the spec says we need to do. its one of the rare cases where trading verbosity may improve readability

Reply:
It's funny, the idea that "any developer can pick up and understand other's code" is the exact same reason why style guides (which specify things like variable naming conventions) are popular.

Typical style guides say that variables with all capitol letters are static constants. So to my brain, when I read the code I ask myself "Why is the Program Counter a constant? What if I want to put a different value in there?".

But alternatively I understand your point about trying to match the design specs as closely as possible. I see things like this as a difference of style preference, and while I wish we all did the /obviously/ best style (my style 😎 ), I understand that everyone has different preferences so I'll be flexible.

Reply:
haha, no i totally get you, i think i lowercased all of them in one of the last couple commits. imo its good enough to match the acronym, capitalizing is over the top, even for me lol